### ISSUE TITLE
[bitnami/mongodb] Can't upgrade chart when auth.enable=false

### ISSUE THREAD

**Issue by obeyda** [bitnami/charts#3425]
Which chart:
bitnami/mongodb@8.3.0

**Describe the bug**
when I install the chart with `auth.enabled=false`, the I try to upgrade the chart (any value) I get an error

**To Reproduce**
1. install the chart with `--set auth.enabled=false`
2. update any value (the service values for example) 
3. upgrade
4. see error
```
UPGRADE FAILED
Error: render error in "mongodb/templates/NOTES.txt": template: mongodb/templates/NOTES.txt:190:4: 
executing "mongodb/templates/NOTES.txt" at <include "common.errors.upgrade.passwords.empty" 
(dict "validationErrors" (list $requiredPasswordValidationErrors) "context" $)>: error calling include:
 template: mongodb/charts/common/templates/_errors.tpl:18:48: executing "common.errors.upgrade.passwords.empty" 
 at <fail>: error calling fail: 
PASSWORDS ERROR: you must provide your current passwords when upgrade the release
    'auth.rootPassword' must not be empty, please add '--set auth.rootPassword=$MONGODB_ROOT_PASSWORD' to the command. To get the current value:

        export MONGODB_ROOT_PASSWORD=$(kubectl get secret --namespace namespace my-mongodb -o jsonpath="{.data.mongodb-root-password}" | base64 --decode)

    'auth.replicaSetKey' must not be empty, please add '--set auth.replicaSetKey=$MONGODB_REPLICA_SET_KEY' to the command. To get the current value:

        export MONGODB_REPLICA_SET_KEY=$(kubectl get secret --namespace namespace my-mongodb -o jsonpath="{.data.mongodb-replica-set-key}" | base64 --decode)

ROLLING BACK
```

**Expected behavior**

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
Client: &version.Version{SemVer:"v2.16.7", GitCommit:"5f2584fd3d35552c4af26036f0c464191287986b", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.1", GitCommit:"bbdfe5e7803a12bbdf97e94cd847859890cf4050", GitTreeState:"clean"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.5", GitCommit:"e6503f8d8f769ace2f338794c914a96fc335df0f", GitTreeState:"archive", BuildDate:"2020-07-01T16:28:46Z", GoVersion:"go1.14.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.4", GitCommit:"8a6cdee2dab2329284444e9448bdf8ab165e1c5c", GitTreeState:"clean", BuildDate:"2020-06-24T18:14:29Z", GoVersion:"go1.13.6", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by javsalgar** [comment]
Hi!

Thank you very much for reporting. I believe that the password check should not be executed when auth.enabled=false. We will review the current logic.

**Comment by javsalgar** [comment]
I created this PR for this use case https://github.com/bitnami/charts/pull/3449/files

**Comment by obeyda** [comment]
@javsalgar, I am still having this issue with the latest mongodb chart: 9.0.0
can you please reopen?

**Comment by obeyda** [comment]
I've create a PR to fix this: #3544

**Comment by javsalgar** [comment]
Thanks! we will review the PR
